xtext.cpp

```diff
@@ -360,7 +360,7 @@ class XText {
         if (m_bidi_ctypes) { free(m_bidi_ctypes); m_bidi_ctypes = NULL; }
         if (m_bidi_btypes) { free(m_bidi_btypes); m_bidi_btypes = NULL; }
         if (m_bidi_levels) { free(m_bidi_levels); m_bidi_levels = NULL; }
-        if (m_lang)        { delete m_lang;       m_lang = NULL; }
+        if (m_lang)        { delete[] m_lang;     m_lang = NULL; }
```

Comment:
Does this change anything in any way in what the compiler would do ? Or it's just for sake of correctness?

Reply:
It deletes the entire array of characters instead of the first one. Until now the memory of the rest of characters was leaked on each deallocation

Reply:
I get the idea.
But it's one single thing I created/allocated with `m_lang = new char[strlen(lang)+1];` - so I find it strange the compiler would just delete the first char (a fragment of that thing, while it knows how much was allocated). It's my idea the compiler would do the right thing, but I don't know him personally :)

Reply:
>  I don't know him personally :)

Me neither :smile: 

But C++ is a humongous beast with operator overloading and other stuff and this specific behaviour is forbidden. See https://isocpp.org/wiki/faq/freestore-mgmt#delete-array-built-ins

Reply:
It's *apparently* mostly harmless on Linux, but, it is admittedly an implementation detail, and Valgrind was certainly unhappy about it ;).